src/js/apps/patients/worklist/worklist_app.js

```diff
@@ -63,27 +68,42 @@ export default App.extend({
     'change': 'onChangeState',
   },
   onChangeState(state) {
+    store.set(this.stateId, state);
     if (state.hasChanged('groupId')) {
       this.restart();
     }
   },
   viewEvents: {
     'toggle:listType': 'onToggleListType',
   },
+  initListState() {
+    if (!this.stateId) {
```

Comment:
what's the use case for the else here?  when is `stateId` defined by this point?

Reply:
When the app is restarted, `stateId` remains defined, but `worklistId` and `worklistType` end up undefined in `onBeforeStart` because options is undefined. So if we don't have that check, `stateId` gets overwritten with undefined values.

Reply:
if the app is restarted I don't believe this function is called at all.

Reply:
Oh geez. Sorry I see what happened. Originally that `stateId` logic was in `onBeforeStart` above the `isRestarting` check. You're right, we don't need that anymore